// We're using proto2 syntax
syntax = "proto2";

package pridict;

service Reccomandation {
  rpc Predict(queryRequest) returns (queryResponse) {}
  rpc SimillarItem(simillarItemRequest) returns (simillarItemResponse) {}
  rpc SimillarUser(simillarUserRequest) returns (simillarUserResponse) {}
}

message queryRequest {
  required int32  user = 1;
  required int32  item = 2;
}


message queryResponse {
  required int64 result = 1;
}

message simillarUserRequest {
  required int32  user = 1;
}


message simillarUserResponse {
  required int64 result = 1;
}

message simillarItemRequest {
  required int32  item = 1;
}


message simillarItemResponse {
  required int64 result = 1;
}